docs/SQL/development/visual-editor/variant-schema.md

```diff
@@ -0,0 +1,85 @@
+Using the variant schema functionality, you can do the following:
+
+- Parse JSON and XML formats
+- Infer the variant schema
+- Configure the parsing limit for inferring the column structure
+- Use a nested column inside of the Visual Expression Builder
+
+## Inferring the variant schema
+
+Variant schemas are not stored within the table definition and can vary for each row, making them difficult to infer and use. Fortunately, you don't have to infer the schema yourself. You can use the column selector inside of your Gems to automatically infer the variant schema, explore the multi-type variant structure, and later select a nested column to use in your transformations.
+
+To automatically infer the variant schema, start by opening a Gem that uses a variant column input.
+
+- Select the variant column, and click **Infer Schema**.
+
+Prophecy will automatically detect and identify the variant types in your input data.
+
+![Schema and column selector](img/variant-infer-schema.png)
+
+:::note
+
+The inferred schema is cached so that you can use it again in the future whenever you reopen the Model, Gem, or another Gem connected to the same input port. Check the bottom of the column selector to see the last time the variant schema was inferred.
+
+To refresh the schema, simply click **Infer Schema** again.
+
+:::
+
+After you infer the schema, you can click **Edit Schema** to view the variant schema and make edits to it. Use the Type dropdowns to manually choose the data type of each nested schema.
+
+![Edit schema view](img/variant-edit-schema.png)
+
+You can always use the Code view to write the code query if you want to set the schema yourself.
+
+## Variant sampling setting
+
+When Prophecy infers the variant schema, it samples the records to identify all potential iterations of keys and values within the schema.
+
+The default number of records that Prophecy parses to understand the nested data schema is 100. You can update this limit under the Development Settings, which you can navigate to by clicking **... > Development Settings**.
+
+![Variant sampling setting](img/variant-sampling-setting.png)
+
+We recommend that you increase the limit for small structures, or decrease it for larger ones.
+
+:::note
+
+This setting does not rely on the ratio of the data since that would require a complete count of the data records.
+
+:::
+
+## Adding a nested column
+
+With in the column selector, you can add a nested column by clicking **Add Column** next to the input field name.
+
+![Add column](img/variant-add-column.png)
+
+When adding a column nested within a variant, the output column name, expression, and data type are automatically generated according to the following rules:
+
+- Column name: The column name matches the input field name. If there's a duplicate, the name is prefixed with the next available parent field until it becomes unique.
+
+  For example, if the column name `name` already exists, the new field might be named `customers_name`.
```

Comment:
Actually we always keep the column name as the entire parent path combined and if its conflicting we append `_0` and so on numbers.

Reply:
Done.